examples/pl_domain_templates/generative_adversarial_net.py

```diff
@@ -239,15 +239,8 @@ def main(args: Namespace) -> None:
     cli_lightning_logo()
     parser = ArgumentParser()
 
-    # Add program level args, if any.
-    # ------------------------
-    # Add LightningDataLoader args
-    parser = MNISTDataModule.add_argparse_args(parser)
     # Add model specific args
     parser = GAN.add_argparse_args(parser)
```

Comment:
I suggest taking the code inside this method and putting it here for simplicity. The implementation can also remove the `use_argument_group=False` code path